akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -167,6 +175,7 @@ object GraphStageLogic {
       behaviour = receive
     }
 
+    private[this] var watching = Set.empty[ActorRef]
```

Comment:
Isn't the default set a HashSet? (I'm thinking a TreeSet might be more space efficient)